cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -0,0 +1,482 @@
+   *---------------------------------------------------------------------------**/
+  __host__ __device__ bool nullable() const noexcept {
+    return nullptr != _null_mask;
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the count of null elements
+   *---------------------------------------------------------------------------**/
+  __host__ __device__ size_type null_count() const noexcept {
+    return _null_count;
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Indicates if the column contains null elements,
+   * i.e., `null_count() > 0`
+   *
+   * @return true The column contains null elements
+   * @return false All elements are valid
+   *---------------------------------------------------------------------------**/
+  __host__ __device__ bool has_nulls() const noexcept {
+    return _null_count > 0;
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns raw pointer to the underlying bitmask allocation.
+   *
+   * @note This function does *not* account for the `offset()`.
+   *
+   * @note If `null_count() == 0`, this may return `nullptr`.
+   *---------------------------------------------------------------------------**/
+  __host__ __device__ bitmask_type const* null_mask() const noexcept {
+    return _null_mask;
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the index of the first element relative to the base memory
+   * allocation, i.e., what is returned from `head<T>()`.
+   *---------------------------------------------------------------------------**/
+  __host__ __device__ size_type offset() const noexcept { return _offset; }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns if the specified element holds a valid value (i.e., not
+   * null)
+   *
+   * @note It is undefined behavior to call this function if `nullable() ==
+   * false`.
+   *
+   * @param element_index The index of the element to query
+   * @return true The element is valid
+   * @return false The element is null
+   *---------------------------------------------------------------------------**/
+  __device__ bool is_valid(size_type element_index) const noexcept {
+    return bit_is_set(_null_mask, element_index);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns if the specified element is null
+   *
+   * @note It is undefined behavior to call this function if `nullable() ==
+   * false`.
+   *
+   * @param element_index The index of the element to query
+   * @return true The element is null
+   * @return false The element is valid
+   *---------------------------------------------------------------------------**/
+  __device__ bool is_null(size_type element_index) const noexcept {
```

Comment:
Is it expensive to check `nullable()`? we used to do that in the past without apparent consequence.
Can have another function called `is_null_unsafe` for people who know what they're doing, similar to how we have in old utils. Actually I prefer the suffix: `_nocheck`

Reply:
> Is it expensive to check nullable()

Yes, I recall @harrism saying there was a 30% overhead in an inner loop using `gdf_is_valid` which performs an equivalent check to `nullable()`.

I think you're right that there should be two versions to avoid surprises.